src/parrot_mcp_server/security_core.py

```
@@ -0,0 +1,431 @@
+        start_response(status, [("Content-Type", "application/json"), ("Content-Length", str(len(body)))])
+        return [body]
+
+    def _fire_event(self, result: ThreatResult) -> None:
+        """Schedule the async event callback without blocking the WSGI path."""
+        if self._on_event is None:
+            return
+        import asyncio  # noqa: PLC0415
+
+        try:
+            loop = asyncio.get_running_loop()
+            loop.create_task(self._on_event(result))
```

Ruff (0.15.5)

[warning] Store a reference to the return value of `loop.create_task`

(RUF006)